package.json

```diff
@@ -15,15 +15,20 @@
     "axios": "^1.6.8",
     "bootstrap": "^5.3.3",
     "bootstrap-icons": "^1.11.3",
+    "chart.js": "^4.4.3",
     "jwt-decode": "^4.0.0",
+    "moment": "^2.30.1",
     "react": "^18.2.0",
     "react-bootstrap": "^2.10.2",
+    "react-chartjs-2": "^5.2.0",
     "react-dom": "^18.3.1",
+    "react-donut-chart": "^1.3.3",
     "react-hook-form": "^7.51.4",
     "react-pro-sidebar": "^1.1.0",
     "react-responsive-pagination": "^2.5.0",
     "react-router-dom": "^6.23.1",
     "react-toastify": "^10.0.5",
+    "use-local-storage": "^3.0.0",
```

Comment:
Depending heavily on external packages is not a good practice. Especially if it is something that we can do like creating a custom hook for retrieving and updating local storage.

I noticed also that we have two packages for charts. If one  of them is unused, we need to remove it.

Reply:
Also in some cases, those packages might break your application or become deprecated.
And if someone else is viewing your code and see many dependencies for small task, he might wonder of what we are capable of.

Reply:
for the 2 packages yes it is a bit conflict and i am going to delete the unused one

Reply:
about localStorage i got it and we will fo the enhacement .. already we were using the browser localstorage not the external one